build/deploy/cockroach.sh

```diff
@@ -1,10 +1,232 @@
+  $sql_entrypoint "$@"
+}
+
+# create_defaultdb is to create a default database. If there doesn't exist a database
+# with the given name, create a new database. Otherwise, no-op.
+create_defaultdb() {
+  defaultdb_name=$1
+  # Check if this database already exists.
+  dbAlreadyExists="$(run_sql_query -e "SELECT 1 FROM [SHOW DATABASES] WHERE database_name='$defaultdb_name'")"
+  # If there's no database with the given name, dbAlreadyExists should have only 1 row.
+  # Otherwise, it should have at lease 2 rows.
+  if [ "$(echo -n "$dbAlreadyExists" | grep -c '^')" -eq 1 ]; then
+    # Create a database with the given name.
+    run_sql_query -e "CREATE DATABASE $defaultdb_name"
+    echo "Finished creating default database $defaultdb_name"
+  else
+    echo "$defaultdb_name already exists"
+    fi
+}
+
+# process_init_files run all the init scripts from /docker-entrypoint-initdb.d.
+# This is largely based on PostgreSQL's docker-entrypoint.sh:
+# https://github.com/docker-library/postgres/blob/master/14/alpine/docker-entrypoint.sh#L144-L173
+#
+# usage: process_init_files [file [file [...]]]
+# e.g. process_init_files /your_folder/*
+process_init_files() {
+	for f in "$@"; do
+		case "$f" in
+			*.sh)
+				if [ -x "$f" ]; then
+					echo "$0: running $f"
+					"$f"
+				else
+					echo "$0: sourcing $f"
+					. "$f"
+				fi
+				;;
+			*.sql)    echo "$0: running $f"; run_sql_query -f "$f"; echo ;;
+			*.sql.gz) echo "$0: running $f"; gzip -cd "$f" | run_sql_query; echo ;;
+			*.sql.xz) echo "$0: running $f"; xz --decompress --stdout "$f" | run_sql_query; echo ;;
+			*)        echo "$0: ignoring $f" ;;
+		esac
+		echo
+	done
+}
+
+# setup_db is to create a default database, create a default user (with password if applicable),
+# and grant privileges to this user by running `cockroach sql` queries.
+setup_db() {
+    create_defaultdb $COCKROACH_DATABASE
+
+    local init_env_query=( )
+
+    local create_user_query="CREATE USER $COCKROACH_USER"
+
+    # If COCKROACH_PASSWORD is not empty, create this default user with given password.
+    if [[ ! -z "$COCKROACH_PASSWORD" ]]; then
+          create_user_query+="WITH PASSWORD $COCKROACH_PASSWORD"
+    fi
+
+    init_env_query+=( -e "$create_user_query" \
+                      -e "GRANT ALL ON DATABASE $COCKROACH_DATABASE TO $COCKROACH_USER" \
+                      -e "GRANT admin TO $COCKROACH_USER " )
+
+    # Check if the given command contains the `--insecure` flag.
+    # If not, add the `--certs-dir=certs` to init_env_query.
+    if [[ "$*" =~ $INSECURE_MODE_REGEX ]]; then
+              init_env_query+=( --insecure )
+              else
+              init_env_query+=( "--certs-dir=$CERTS_DIR" )
+    fi
+
+    run_sql_query "${init_env_query[@]}"
+}
+
+# start_init_node is to start the single node for the initialization.
+start_init_node() {
+  echo "Starting node for the initialization process. This could take couple seconds"
```

Comment:
is this is missing a `.` at the end for example? Or maybe we should use `...` when there is a wait.

sometimes we capitalise in the error/echo messages and sometimes we don't. which way should we go?

Reply:
Following the cockroach convention, I changed it to a lowercase start, and added the `...`.